app/code/Magento/Catalog/Model/Product/Attribute/Repository.php

```diff
@@ -235,7 +236,7 @@ protected function generateCode($label)
      */
     protected function validateCode($code)
     {
-        $validatorAttrCode = new \Zend_Validate_Regex(['pattern' => '/^[a-z][a-z_0-9]{0,30}$/']);
+        $validatorAttrCode = new \Zend_Validate_Regex(['pattern' => '/^[a-zA-Z][a-zA-Z_0-9]{0,' . Attribute::ATTRIBUTE_CODE_MAX_LENGTH . '}$/']);
```

Comment:
I believe we should accept just lower case characters here, as described on the docs:

https://docs.magento.com/user-guide/stores/attributes-product.html
```
Attribute Code | (Required) A unique identifier for internal use. The Attribute Code must begin with a letter, but can include a combination of lowercase letters (a-z) and numbers (0-9). The code must be less than thirty characters in length and cannot include any special characters or spaces, although an underscore (_) can be used to indicate a space.
```

Reply:
done